### PR TITLE
Support URL to URL mappings

### LINKED ISSUE #97: Support URL to URL mappings
URL to URL mappings are currently supported in the import map - per the spec support for this should be added.

### PR DESCRIPTION
**Pull request by vovacodes**
Fixes #97

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -151,13 +151,13 @@ export function resolveUrl (relUrl, parentUrl) {
...
-function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap) {
+function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap, parentUrl) {
   for (let p in packages) {
-    const resolvedLhs = resolveIfNotPlainOrUrl(p, baseUrl) || p;
+    const resolvedLhs = resolveIfNotPlainOrUrl(p, parentUrl != undefined ? parentUrl : baseUrl) || p;
     let target = packages[p];
     if (typeof target !== 'string') 
       continue;
-    const mapped = resolveImportMap(parentMap, resolveIfNotPlainOrUrl(target, baseUrl) || target, baseUrl);
+    const mapped = resolveImportMap(parentMap, resolveIfNotPlainOrUrl(target, baseUrl) || target, parentUrl);
```
Comment (guybedford) [inline src/common.js:160]: The target itself must always be resolved relative to the `baseUrl` not the scope Url.

File: src/common.js [hunk 2]
```diff
@@ -151,13 +151,13 @@ export function resolveUrl (relUrl, parentUrl) {
   return resolveIfNotPlainOrUrl(relUrl, parentUrl) || (relUrl.indexOf(':') !== -1 ? relUrl : resolveIfNotPlainOrUrl('./' + relUrl, parentUrl));
 }
 
-function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap) {
+function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap, parentUrl) {
```
Comment (guybedford) [inline src/common.js:154]: Can we call this `scopeUrl`?

File: src/common.js [hunk 3]
```diff
@@ -151,13 +151,13 @@ export function resolveUrl (relUrl, parentUrl) {
...
   for (let p in packages) {
-    const resolvedLhs = resolveIfNotPlainOrUrl(p, baseUrl) || p;
+    const resolvedLhs = resolveIfNotPlainOrUrl(p, parentUrl != undefined ? parentUrl : baseUrl) || p;
```
Comment (guybedford) [inline src/common.js:156]: If we are happy with `!=` semantics then `scopeUrl || baseUrl` is also fine :)

**Review by guybedford — APPROVED** [review]
Perfect, that's the fix - thanks!

File: src/common.js (hunk 3 above)
Comment (vovacodes) [reply src/common.js:156]: `scopeUrl || baseUrl` wouldn't will return `baseURL` id `scopeURL` is an empty string, I'm not sure if an empty string can occur in that position though

Comment (guybedford) [reply src/common.js]: Good point. Yes best not to fret over these details so happy to leave as-is. An empty string is permitted, but the assumption in this code here is that the URLs are already normalized so should never be the empty string I believe.

**Pull request merged**